pkg/apis/v1alpha1/cluster_supply_chain.go

```diff
@@ -254,27 +144,58 @@ type SupplyChainResource struct {
 	Configs []ResourceReference `json:"configs,omitempty"`
 }
 
-var ValidSupplyChainTemplates = []client.Object{
-	&ClusterSourceTemplate{},
-	&ClusterImageTemplate{},
-	&ClusterConfigTemplate{},
-	&ClusterTemplate{},
-}
-
 type SupplyChainTemplateReference struct {
 	// Kind of the template to apply
 	//+kubebuilder:validation:Enum=ClusterSourceTemplate;ClusterImageTemplate;ClusterTemplate;ClusterConfigTemplate
 	Kind string `json:"kind"`
+
+	// Name of the template to apply
+	// Only one of Name and Options can be specified.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+
+	// Options is a list of template names and Selectors. The templates must all be of type Kind.
+	// A template will be selected if the workload matches the specified Selector.
+	// Only one template can be selected.
+	// Only one of Name and Options can be specified.
+	// +kubebuilder:validation:MinItems=2
```

Comment:
Given that the `minitems=2` constraint doesn't show up in `explain`

```
KIND:     ClusterSupplyChain
VERSION:  carto.run/v1alpha1

RESOURCE: options <[]Object>

DESCRIPTION:
     Options is a list of template names and Selectors. The templates must all
     be of type Kind. A template will be selected if the workload matches the
     specified Selector. Only one template can be selected. Only one of Name and
     Options can be specified.

FIELDS:
   name <string> -required-
     Name of the template to apply

   selector     <Object> -required-
     Selector is a field query over a workload resource.
```

wdyt of including an entry that states such `minitems=2` constraint? e.g.

```
// A minimum of two options must be specified
```